### ISSUE TITLE
Support python 3.12

### ISSUE THREAD

**Issue by baseplate-admin** [flet-dev/serious-python#54]
HI,

Thanks for your work on `serious-python`, seriously it's awesome.

I was wondering if it was possible to add Python 3.12 support for `serious-python`

**Comment by FeodorFitsner** [comment]
Working on that right now.

**Comment by baseplate-admin** [comment]
Hi, thanks for your hard work.

I have a small question..

Would `serious-python` move from `python-from-android`  toolchain?

**Comment by FeodorFitsner** [comment]
Yes, we are going to move from `python-from-android` toolchain to the one based on https://github.com/beeware/mobile-forge which will work with both iOS and Android.

**Comment by baseplate-admin** [comment]
Hi @FeodorFitsner, 

could we make flet support many python versions? For example 3.10-3.12?

**Comment by syleishere** [comment]
I think 3.11 up should be good. On windows, WSL, ubuntu no issues with 3.12. On FreeBSD still issues for development team moving from 3.11 currently.

**Comment by baseplate-admin** [comment]
> I think 3.11 up should be good

I think otherwise. We can reap the benefits of [Faster CPython](https://github.com/faster-cpython/) project that way.

**Comment by Paras-Shah-sudo** [comment]
Hello there, I was working on an open source project that uses Python3.7.16 specifically and I wanted to convert it into an apk. Since there is currently no support on changing the host python version used for packaging the app, I was wondering if I could modify the code to work for my particular requirement. So I was hoping someone could tell me where to look in the code base which defines how the host python is selected. Maybe I could point it to my anaconda environment and it also gets all the libraries installed in that environment and it works...

Thanks

**Comment by ndonkoHenri** [comment]
Long since done. And better, today, serious_python bundles three Python lines side by side, currently **3.12.13 / 3.13.14 / 3.14.6** as of 4.5.1. 3.12 first shipped in 0.9.0; [multi-version selection](https://flet.dev/blog/flet-v-0-86-release-announcement#multi-version-python) landed in 2.0.0.

Pick one at packaging time:

```bash
dart run serious_python:main package app/src -p Android --python-version 3.12
```

(or `export SERIOUS_PYTHON_VERSION=3.12`). The default since 2.0.0 is the latest stable, currently 3.14 ([resolution in `flet build`](https://flet.dev/docs/publish/#choosing-a-python-version)).

@sudo-paras-shah on the 3.7 question — only the versions in that table are available. The `package` command downloads its own host CPython (from [python-build-standalone](https://github.com/astral-sh/python-build-standalone)) and pairs it with the matching embedded runtime from [flet-dev/python-build](https://github.com/flet-dev/python-build); it can't be pointed at a local or conda environment, and no 3.7 runtime exists for the target platforms.
